app/components/primer/select_menu_component.rb

```diff
@@ -0,0 +1,499 @@
+
+      # @param selected [Boolean] Whether this item is the currently active one.
+      # @param tab [Integer] Which tab this item should appear in. The first tab is 1.
+      # @param icon [String] Octicon name for this item. Defaults to no icon. Set to a value like `"check"` to add an [Octicon](https://primer.style/octicons/) to this item.
+      # @param divider [Boolean, String, nil] Whether to show a divider after this item. Pass `true` to show a simple line divider, or pass a String to show a divider with a message.
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>, including: `tag` (`Symbol`) - HTML element type for the item tag; defaults to `:button`. `role` (`String`) - HTML role attribute for the item tag; defaults to `"menuitem"`. `divider_classes` (`String`) - CSS classes to apply to the divider after this item; only used if `divider` is not `nil`. `icon_classes` (`String`) - CSS classes to apply to the icon; only used if `icon` is not `nil`.
+      def initialize(selected: DEFAULT_SELECTED, icon: nil, tab: DEFAULT_TAB, divider: nil, **system_arguments)
+        @selected = fetch_or_fallback_boolean(selected, DEFAULT_SELECTED)
+        @icon = icon
+        @tab = (tab || DEFAULT_TAB).to_i
+        @divider = divider
+        @system_arguments = system_arguments
+        @system_arguments[:tag] ||= :button
+        @system_arguments[:role] ||= if @selected || @icon
+          "menuitemcheckbox"
+        else
+          "menuitem"
+        end
+        @system_arguments[:classes] = class_names(
+          "SelectMenu-item",
+          system_arguments[:classes],
+        )
+        @system_arguments[:"aria-checked"] = "true" if @selected
+      end
+
+      def wrapper_component
+        case @system_arguments[:tag]
+        when :button
+          Primer::ButtonComponent.new(**@system_arguments)
+        when :a
+          Primer::LinkComponent.new(**@system_arguments)
+        else
+          Primer::BaseComponent.new(**@system_arguments)
+        end
+      end
+
+      # Private: Only used if `icon` is non-nil.
+      def icon_component
+        Primer::OcticonComponent.new(
+          icon: icon,
+          classes: class_names(
+            "SelectMenu-icon SelectMenu-icon--check",
+            @system_arguments[:icon_classes],
+          )
+        )
+      end
+
+      # Private: Only used if `divider` is non-nil.
+      def divider_component
+        divider_classes = class_names(
+          "SelectMenu-divider",
+          @system_arguments[:divider_classes],
+        )
+        if divider.is_a?(String)
+          Primer::BaseComponent.new(
+            tag: :div,
+            classes: divider_classes
+          )
+        else
+          Primer::BaseComponent.new(
+            tag: :hr,
+            classes: divider_classes
+          )
+        end
+      end
+    end
+
+    # An optional header for the select menu.
+    class Header < Primer::Slot
+      DEFAULT_CLOSE_BUTTON = false
```

Comment:
What do you think of the term `closable`?

```suggestion
      DEFAULT_CLOSEABLE = false
```